### PR TITLE
Update mqtt message format

### PR DESCRIPTION
**Pull request by ikstream**
This pr implements the proposed new JSON message format
#### Added
```
optional meta data object
moving t5060 to optinal object
adding optional t1012 object
adding package counter
```

#### Removed
iThe old event mqtt code was removed, the new event object
is not implemented yet.

#### Fixed
This pr includes a fix for uci config read stopping and leaving us with an incomplete conf struct if an option was not set in /etc/config/powquty. For additional information please see the according commit

#### Updated
The Readme was updated to represent the new message format and provide configuration file options

### CODE REVIEW

File: powqutyd/files/src/uci_config.c [hunk 1]
```diff
@@ -28,6 +28,25 @@ static long uci_lookup_option_long(struct uci_context *uci,
 	return str == NULL ? -1 : atol(str);
 }
 
+/*
+ * check option value from uci config
+ * @param option_value: name of the option to check
+ * @param option_length: max length of option value
+ * @return 1 on to long option value, -1 if unset, 0 else
+ */
+static int check_option_value(char * option_value, unsigned int option_length) {
```
Comment (thuehn) [inline powqutyd/files/src/uci_config.c]: should we change the function name  to be more clear ?
I would prefer something like: 
  static bool config_option_is_valid() ... 0 = failure, non-zero = success

File: powqutyd/files/include/mqtt.h [hunk 2]
```diff
@@ -10,8 +10,8 @@
 
 #include "PQ_App.h"
 
-#define MAX_MQTT_MSG_LEN		400
-#define mqtt_port 1883
+#define MAX_MQTT_MSG_LEN	1024
+#define mqtt_port		1883
```
Comment (thuehn) [inline powqutyd/files/include/mqtt.h]: Should'nt we have this MARCO "mqtt_port" in upper case : "MQTT_PORT"

File: powqutyd/files/src/mqtt.c [hunk 3]
```diff
@@ -129,6 +130,27 @@ void mqtt_message_print(struct mosquitto_message* msg) {
...
+			"\"operator\": \"%s\", "
+			"\"phase\": \"%s\", "
+			"\"reason\": \"%s\", "
+			"\"type\": \"%s\"",
+			conf->meta_comment,
+			conf->meta_id,
+			conf->meta_operator,
+			conf->meta_phase,
+			conf->meta_reason,
+			conf->meta_type);
```
Comment (thuehn) [inline powqutyd/files/src/mqtt.c]: plz check indentation, I think all line need to be alined below the "m"etadata

File: powqutyd/files/src/uci_config.c [hunk 4]
```diff
@@ -158,10 +202,84 @@ int uci_config_powquty(struct powquty_conf* conf) {
...
+			if (str == NULL)
+				continue;
+			if (strlen(str) >= MAX_LONG_LENGTH - 1) {
+				continue;
+			}
+			strcpy(conf->meta_operator, str);
+			printf("looking up meta_operator: currently ==> %s\n",
+				conf->meta_operator);
+
+			/* pahse */
```
Comment (thuehn) [inline powqutyd/files/src/uci_config.c]: typo ?

File: powqutyd/files/src/mqtt.c [hunk 5]
```diff
@@ -17,6 +17,9 @@
 #include <stdlib.h>
 #include <errno.h>
 
+#define STATIC_DATA_LENGTH 210
```
Comment (thuehn) [inline powqutyd/files/src/mqtt.c]: out of curiosity: where comes the 210 from ?

File: powqutyd/files/src/mqtt.c [hunk 6]
```diff
@@ -141,12 +139,14 @@ void mqtt_message_print(struct mosquitto_message* msg) {
 void compose_metadata(struct powquty_conf* conf) {
 	metadata[0] = '\0';
 	sprintf(metadata,
+			"\"metadata\": {"
```
Comment (thuehn) [inline powqutyd/files/src/mqtt.c]: right indentation ?

File: powqutyd/files/include/mqtt.h (hunk 2 above)
Comment (ikstream) [reply powqutyd/files/include/mqtt.h]: I did not implement that macro and didn't want to change it, but can do so.

File: powqutyd/files/src/uci_config.c (hunk 4 above)
Comment (ikstream) [reply powqutyd/files/src/uci_config.c]: yes

File: powqutyd/files/src/mqtt.c (hunk 5 above)
Comment (ikstream) [reply powqutyd/files/src/mqtt.c]: I don't have the paper in front of me atm, but I calculated how many character where possible with in the config file, added the JSON characters and, i think, 2character to be sure i did not leave anything out.
```
-five variables with max length of 32 character
-somewhere around 40 JSON string character
```

**Comment by thuehn** [comment]
you are the man! ... lets put this online.

Thx & Merged

**Pull request merged**